src/cluster/managers/documentation/CommandDocumentationManager.ts

```diff
@@ -0,0 +1,225 @@
+        const description = [];
+        switch (result.state) {
+            case 'ALLOWED':
+                break;
+            case 'BLACKLISTED':
+            case 'DISABLED':
+            case 'NOT_IN_GUILD':
+            case 'MISSING_PERMISSIONS':
+            case 'MISSING_ROLE':
+                description.push(codeBlock(`❌ You cannot use b!${result.detail.command.name}`));
+                break;
+        }
+        const { detail: { command } } = result;
+        const fields: EmbedField[] = [];
+        if (command.description !== undefined && command.description.length > 0)
+            description.push(command.description);
+
+        if (command.aliases.length > 0)
+            fields.push({ name: '**Aliases**', value: command.aliases.join(', ') });
+
+        if (command.flags.length > 0)
+            fields.push({ name: '**Flags**', value: humanize.flags(command.flags).join('\n') });
+
+        const signatures = command.signatures
+            .filter(c => !c.hidden)
+            .map(signature => ({
+                usage: humanize.commandParameters(signature.parameters),
+                description: signature.description,
+                notes: signature.parameters.flatMap(p => [...this.#getParameterNotes(p)])
+            }));
+
+        const pages: DocumentationPage[] = [];
+        for (const signature of signatures) {
+            const usage = `b!${command.name}${signature.usage !== '' ? ` ${signature.usage}` : ''}`;
+            pages.push({
+                name: usage,
+                embed: {
+                    fields: [
+                        ...fields,
+                        {
+                            name: `ℹ️  ${usage}`,
+                            value: `${signature.notes.map(n => `> ${n}`).join('\n')}\n\n${signature.description}`.trim()
+                        }
+                    ]
+                }
+            });
+        }
+
+        return {
+            id: command.id,
+            name: command.name,
+            type: 'paged',
+            hidden: result.state !== 'ALLOWED',
+            tags: [command.name, ...command.aliases],
+            embed: {
+                url: `/commands#${command.name}`,
+                description: description.join('\n'),
+                color: this.#getColor(command.category)
+            },
+            selectText: 'Pick a command signature',
+            pages: pages.sort((a, b) => a.name < b.name ? -1 : 1)
+        };
+    }
+
+    #getColor(type: string): number {
+        switch (type.toLowerCase()) {
+            case 'custom': return 0x7289da;
+            case 'general': return 0xefff00;
+            case 'nsfw': return 0x010101;
+            case 'image': return 0xefff00;
+            case 'admin': return 0xff0000;
+            case 'social': return 0xefff00;
+            case 'owner': return 0xff0000;
+            case 'developer': return 0xff0000;
+            case 'staff': return 0xff0000;
+            case 'support': return 0xff0000;
```

Comment:
I think these colors are already duplicated in an enum or something. Might be worth taking a look at other files, right?

Reply:
They are in a few places. I think doing a color consolidtation pass would be best left to another PR though